<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ExtractCenterline</class>
 <widget class="qMRMLWidget" name="ExtractCenterline">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>504</width>
    <height>863</height>
   </rect>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="parameterSetLabel">
     <property name="text">
      <string>Paremeter set: </string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="qMRMLNodeComboBox" name="parameterNodeSelector">
     <property name="toolTip">
      <string>Pick node to store parameter set</string>
     </property>
     <property name="nodeTypes">
      <stringlist>
       <string>vtkMRMLScriptedModuleNode</string>
      </stringlist>
     </property>
     <property name="showHidden">
      <bool>true</bool>
     </property>
     <property name="baseName">
      <string>ExtractCenterline</string>
     </property>
     <property name="renameEnabled">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Probe Placement</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="rightMargin">
       <number>11</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Probe Segment:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Place Probes:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="qMRMLNodeComboBox" name="endPointsMarkupsSelector">
          <property name="toolTip">
           <string>Branch endpoints. &quot;Unselected&quot; control points are used as sources, &quot;selected&quot; control points are used as targets.</string>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLMarkupsFiducialNode</string>
           </stringlist>
          </property>
          <property name="showChildNodeTypes">
           <bool>false</bool>
          </property>
          <property name="baseName">
           <string>Endpoints</string>
          </property>
          <property name="noneEnabled">
           <bool>true</bool>
          </property>
          <property name="addEnabled">
           <bool>true</bool>
          </property>
          <property name="removeEnabled">
           <bool>true</bool>
          </property>
          <property name="editEnabled">
           <bool>true</bool>
          </property>
          <property name="renameEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qSlicerMarkupsPlaceWidget" name="endPointsMarkupsPlaceWidget"/>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <widget class="ctkPushButton" name="PushButton">
        <property name="text">
         <string>Place Probes</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="probeNodeSelector">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLSegmentationNode</string>
         </stringlist>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="outputsCollapsibleButton">
     <property name="text">
      <string>Tumor Orientation and Margin Evaluation</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Tumor Segment:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qMRMLNodeComboBox" name="tumorSegmentSelector">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLSegmentationNode</string>
           </stringlist>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Native Fiducials:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qMRMLNodeComboBox" name="nativeFiducialsSelector">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLMarkupsFiducialNode</string>
           </stringlist>
          </property>
          <property name="showHidden">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qSlicerMarkupsPlaceWidget" name="MarkupsPlaceWidget"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>New Fiducials:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qMRMLNodeComboBox" name="newFiducialSelector">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLMarkupsFiducialNode</string>
           </stringlist>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qSlicerMarkupsPlaceWidget" name="MarkupsPlaceWidget_2"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_2">
        <property name="text">
         <string>Align Volumes and Translate Tumor</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_7">
        <property name="text">
         <string>Harden Tumor Transformation</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_6">
        <property name="text">
         <string>Translate Probes</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_3">
        <property name="text">
         <string>Evaluate Tumor Margins</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_4">
        <property name="text">
         <string>Apply Margin Color</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkPushButton" name="PushButton_5">
        <property name="text">
         <string>Revert Color</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="7" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkPushButton</class>
   <extends>QPushButton</extends>
   <header>ctkPushButton.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerMarkupsPlaceWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerMarkupsPlaceWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>parameterNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>28</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>192</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>endPointsMarkupsSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>4</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>endPointsMarkupsPlaceWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>4</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>endPointsMarkupsSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>endPointsMarkupsPlaceWidget</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MarkupsPlaceWidget_2</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>nativeFiducialsSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>233</x>
     <y>450</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>newFiducialSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nativeFiducialsSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>MarkupsPlaceWidget</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>450</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>450</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newFiducialSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>MarkupsPlaceWidget_2</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>505</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>tumorSegmentSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>361</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>probeNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>probeNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>probeNodeSelector</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractCenterline</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MarkupsPlaceWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
